Framework/PythonInterface/plugins/algorithms/IntegratePeaksShoeboxTOF.py

```diff
@@ -0,0 +1,687 @@
+                sigma = sigma * bin_width
+            set_peak_intensity(peak, intens, sigma, do_lorz_cor)
+
+        if len(ipks_strong):
+            for weak_pk in weak_peaks_list:
+                # get peak
+                ipk = weak_pk.ipk
+                peak = peaks.getPeak(ipk)
+                bank_name = peaks.column("BankName")[ipk]
+                pk_tof = peak.getTOF()
+                # find nearest strong peak to get shoebox dimensions from
+                ipks_near_strong = []
+                for ipk_near in weak_pk.ipks_near:
+                    if results[ipk_near].status == PEAK_STATUS.STRONG:
+                        ipks_near_strong.append(int(ipk_near))
+                if not ipks_near_strong:
+                    # no peaks in detector window at any TOF, look in all table
+                    ipks_near_strong = ipks_strong
+                # loop over strong peaks and find nearest peak
+                ipk_strong = None
+                angle_min = np.inf
+                for ipk_near in ipks_near_strong:
+                    angle = calc_angle_between_peaks(peak, peaks.getPeak(ipk_near))
+                    if angle < angle_min:
+                        angle_min = angle
+                        ipk_strong = ipk_near
+                strong_pk = peaks.getPeak(ipk_strong)
+                # get peak shape and make kernel
+                nrows, ncols, nbins = results[ipk_strong].peak_shape
+                # scale TOF extent
+                if get_nbins_from_b2bexp_params:
+                    # scale TOF extent by ratio of fwhm
+                    strong_pk_fwhm = get_fwhm_from_back_to_back_params(strong_pk, ws, strong_pk.getDetectorID())
+                    ratio = weak_pk.tof_fwhm / strong_pk_fwhm
+                else:
+                    # scale assuming dTOF/TOF = const
+                    ratio = pk_tof / strong_pk.getTOF()
+                nbins = max(3, round_up_to_odd_number(int(nbins * ratio)))
+                kernel = make_kernel(nrows, ncols, nbins)
+                # get data array in peak region (keep same window size, nshoebox, for plotting)
+                peak_data = array_converter.get_peak_data(
+                    peak, peak.getDetectorID(), bank_name, nshoebox * kernel.shape[0], nshoebox * kernel.shape[1], nrows_edge, ncols_edge
+                )
+                x, y, esq, ispecs = get_and_clip_data_arrays(ws, peak_data, pk_tof, ispec, kernel, nshoebox)
+                # integrate at previously found ipos
+                ipos = [*np.argwhere(ispecs == weak_pk.ispec)[0], np.argmin(abs(x - weak_pk.tof))]
+                det_edges = peak_data.det_edges if not integrate_on_edge else None
+                intens, sigma, status = integrate_shoebox_at_pos(y, esq, kernel, ipos, weak_peak_threshold, det_edges)
+                # scale summed intensity by bin width to get integrated area
+                intens = intens * bin_width
+                sigma = sigma * bin_width
```

Comment:
Also, this final weak peak integration part could be regroup into a separate function?

Reply:
See comment above!